source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -276,6 +280,18 @@ bool VaryHeader::hasVary(const Http::ResponseHeaderMap& headers) {
   return !vary_header.empty() && !vary_header[0]->value().empty();
 }
 
+absl::btree_set<absl::string_view>
```

Comment:
Does this need to be a btree_set? Looks like the callers only iterate over results, perhaps a vector would fit better (and cheaper to maintain?)

Reply:
My thought was that it should deduplicate vary entries. It could be a vector if that's not important.

Reply:
I'm not sure how important it is to de-duplicate header values here. Sgtm if you think it's worth it.